code/manifest.json

```diff
@@ -30,7 +30,7 @@
     "js/inject/hoopla_inject.js",
     "js/controllers/*"
   ],
-  "permissions": ["tabs", "storage", "http://*/*", "https://*/*"],
+  "permissions": ["tabs", "storage", "notifications", "http://*/*", "https://*/*"],
```

Comment:
This needs to use optional permissions. If you just add a new permission then the extension will prompt _everyone_ (installed users included) that a new permission has been added (and I think will disable the extension until accepted? something bad that we don't want at least :))

Please read through the docs on `permissions` api here: https://developer.chrome.com/extensions/permissions

I think that this will be easy to add since the options page is already setup! When the user toggles notifications on we need to:

1. Check if `notifications` permission is enabled. If it is then proceed
2. If `notifications` is not enabled then invoke `chrome.permissions.request` for `notifications` and then in the callback
call `toggleNotifications`

Something like that should work.